transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -717,7 +755,7 @@ protected boolean doConnect(SocketAddress remoteAddress, SocketAddress localAddr
         return connected;
     }
 
-    private boolean doConnect0(SocketAddress remote) throws Exception {
+    boolean doConnect0(SocketAddress remote) throws Exception {
```

Comment:
elevated access for doConnect0 and writeBytes to package private. behavior is no longer strictly generic for epoll/epoll-stream, as EpollSocketChannel has custom requirements for those methods during fast connect. could be a better way to do this

Reply:
If you expose this method, you should probably add a javadoc explaining the return value.    It will now be API.

Reply:
yea, i was wondering about that. AbstractEpollChannel has a package private constructor, so its still an "internal" API, but not exactly hidden. ill add some docs after i confirm this was the right approach, might be a better way still